Templates/Per Tenant Extension/.github/workflows/UpdateGitHubGoSystemFiles.yaml

```diff
@@ -28,9 +28,52 @@ env:
   ALGoRepoSettings: ${{ vars.ALGoRepoSettings }}
 
 jobs:
+  GetBranches:
+    runs-on: windows-latest
+    name: Get Supported Branches
+    outputs:
+      SupportedBranches: ${{ steps.DetermineBranches.outputs.Result }}
+    steps:
+      - name: Checkout
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        uses: actions/checkout@v4
+
+      - name: Read settings
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        uses: microsoft/AL-Go-Actions/ReadSettings@main
+        with:
+          shell: powershell
+          get: updateALGoBranches
+
+      - name: Get Supported Branches
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        id: GetSupportedBranches
+        uses: microsoft/AL-Go-Actions/GetGitBranches@main
+        with:
+          shell: powershell
+          includeBranchesJson: ${{ env.updateALGoBranches }}
+
+      - name: Determine branches to update
+        id: DetermineBranches
+        run: |
+            if (${{ github.event_name }} -eq 'workflow_dispatch') {
+              Write-Host "Running Update AL-Go System Files on a manual trigger."
+              Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "Result=['${{ github.ref_name }}']"
+            }
+            else {
+              Write-Host "Running Update AL-Go System Files on a schedule."
+              Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "Result=$(${{ steps.GetSupportedBranches.outputs.Branches }})"
+            }
+
   UpdateALGoSystemFiles:
-    needs: [ ]
+    needs: [ GetBranches ]
     runs-on: [ windows-latest ]
+    name: "[${{ matrix.branch }}] Update AL-Go"
```

Comment:
Wouldn't it be better to just apply the changes to all branches - instead of running the entire thing for all branches?
Basically have an array of branches in CommitFromNewFolder instead of just one branch.

Reply:
Different branches might be running on different AL-Go versions and have different AL-Go settings, so it's safer to run the logic separately for every branch.

Reply:
So, if you keep templateUrl empty and/or select to not download latest.
got it.